snuba/web/views.py

```diff
@@ -306,18 +307,37 @@ def dataset_query_view(*, dataset: Dataset, timer: Timer):
     elif http_request.method == "POST":
         body = parse_request_body(http_request)
         _trace_transaction(dataset)
-        return dataset_query(dataset, body, timer)
+        return dataset_query(dataset, body, timer, "legacy")
+    else:
+        assert False, "unexpected fallthrough"
+
+
+@application.route("/<dataset:dataset>/snql", methods=["GET", "POST"])
+@util.time_request("snql")
+def snql_dataset_query_view(*, dataset: Dataset, timer: Timer):
+    if http_request.method == "GET":
+        schema = RequestSchema.build_with_extensions(
+            dataset.get_extensions(), HTTPRequestSettings, "snql"
```

Comment:
This would still allow extensions to be provided by the user and the JSON to be valid. `build_with_extensions` builds a validator that takes all the schemas of the extensions and merges them in the main query jsonschema. 
Is this the intended behavior ?

Reply:
For now, yes. There are still some potential extensions that we would want to allow with SnQL that don't fit into the language spec well. There will be a follow up on this where we clean up extensions that SnQL does/does not allow.

Reply:
ok.

> There are still some potential extensions that we would want to allow with SnQL

Do you know which ones ?

Reply:
So far, `turbo` and `debug`. I'm debating `dataset` as well, since it's awkward to add it to the language spec.

Reply:
ok, got it